src/components/CookieSettings.tsx

```diff
@@ -0,0 +1,135 @@
+import { Dispatch, SetStateAction, useEffect, useState } from 'react';
+import { useCookies } from 'react-cookie';
+import Button from './Button';
+import Modal from './Modal';
+import { Switch } from '@headlessui/react';
+import { LockClosedIcon } from '@heroicons/react/outline';
+import { useCookieSettings } from './CookieBanner';
+
+const CookieSettings = (props: {
+  openSettings: boolean;
+  setOpenSettings:
+    | Dispatch<SetStateAction<boolean>>
+    | Dispatch<SetStateAction<Boolean>>
+    | ((open: Boolean) => void);
+
+  decline: () => void;
+}) => {
+  const { setCookieSettings, cookies } = useCookieSettings();
+
+  // cookie defaults
+  const [essentialCookies, setEssentialCookies] = useState<boolean>(
+    cookies.holaplex_cookies_acceptance === 'true' || true
+  );
+  const [analyticsCookies, setAnalyticsCookies] = useState<boolean>(
+    cookies.holaplex_analytics_cookies_acceptance === 'true' || true
+  );
+  const [preferenceCookies, setPreferenceCookies] = useState<boolean>(
+    cookies.holaplex_preference_cookies_acceptance === 'true' || false
+  );
+
+  console.log(preferenceCookies, analyticsCookies);
+
+  useEffect(() => {
+    setAnalyticsCookies(cookies.holaplex_analytics_cookies_acceptance === 'true');
+    setPreferenceCookies(cookies.holaplex_preference_cookies_acceptance === 'true');
+  }, [cookies]);
+
+  const saveSettings = () => {
+    setCookieSettings({
+      analyticsCookies,
+      preferenceCookies,
+      essentialCookies,
+    });
+    props.setOpenSettings(false);
+  };
+
+  return (
+    <Modal open={props.openSettings} setOpen={props.setOpenSettings} title={'Cookie settings'}>
+      <div className={`flex flex-col gap-2`}>
+        <p className={'m-0 border-b border-gray-700 pb-2 text-sm'}>
+          We use cookies, some of them are essentials, others are optional.{' '}
+          <a
+            target={`_blank`}
+            href={`https://docs.google.com/document/d/12uQU7LbLUd0bY7Nz13-F9cua5Wk8mnRNBlyDzF6gRmo/`}
+            className={`font-bold`}
+          >
+            Learn more
+          </a>
+        </p>
+        <form onSubmit={saveSettings} className={`flex flex-col gap-4`}>
+          <div className={'flex items-center justify-between gap-2'}>
+            <div className={'flex flex-col gap-2'}>
+              <h6 className={'m-0 text-base'}>Essential</h6>
+              <p className={'m-0 text-xs text-gray-300'}>
+                {"Necessary for the website to function properly and can't be deactivated"}
```

Comment:
As a site user, I would want to see what these essential cookies are to decide whether or not I actually want to use the site. Otherwise, the "Essential" is relatively meaningless.